packages/eslint-plugin-stories/src/rules/no-new-components-without-story.ts

```diff
@@ -0,0 +1,45 @@
+const fs = require('fs');
+import type { Rule } from 'eslint';
+
+const failureMessage = `
+  React Components must have a corresponding Storybook story.
+
+  "Storybook" is our component explorer, and a "story" is a component example that serves as a unit for Storybook
+
+  Having a Storybook story for each component allows us to test our components' behavior, accessibility,
+  visual regression, etc. Learn more here:
+  https://go.czi.team/storybook-testing
+`;
+
+/**
+ * To encourage Storybook coverage, we warn developers about React components without stories.
+ */
+const rule: Rule.RuleModule = {
+  create(context) {
+    return {
+      ExportDefaultDeclaration(node) {
+        const filename = context.getFilename();
+        const isComponent =
+          filename.match('.tsx') &&
+          !filename.match('.stories.') &&
+          !filename.match('.spec.');
+
+        if (!isComponent) {
+          return;
+        }
+
+        const storyFilename = `${filename.split('.')[0]}.stories.tsx`;
+        const correspondingStoryExists = fs.existsSync(storyFilename);
+
+        if (!correspondingStoryExists) {
+          context.report({
+            node: node,
+            message: failureMessage,
+          });
+        }
+      },
+    };
+  },
+};
+
+export default rule;
```

Comment:
Couple more things you'll need:
- Add docs to the eslint-plugin-stories/README.md, please
- Re: yarn.lock, you'd commit yarn.lock if you expected there to be changes (such as if you added or updated packages or something). In this case, though, it looks like you got the wrong version of Yarn locally.
   
   What do you see when running `yarn -v`?

Reply:
ah I ran `npm i` in the repo and didn't have yarn installed, that was the problem. Thanks for the guidance and I fixed it

Reply:
Actually, I'm not sure why my yarn.lock is still different.

> What do you see when running yarn -v?
```
yarn --version
3.2.1
```

Reply:
Weird. Yarn v3 is absolutely correct.

Not sure what the deal is. If this happens again sometime we can dig into it.